### PR TITLE
doc: update obsolete ref to api/XDS_PROTOCOL.md

### PR DESCRIPTION
**Pull request by silentdai**
Description:

replace `XDS_PROTOCOL.md` by `:ref: xds_protocol series`

Signed-off-by: Yuchen Dai <silentdai@gmail.com>

Risk Level: None
Testing: None
Docs Changes: fix only
Release Notes: None

### CODE REVIEW

**Comment by silentdai** [comment]
Will update.
Question: What is the benefit of the `ref`?  Is it because of relative path? Or is `ref` validated?

**Comment by mattklein123** [comment]
`:ref:` is validated.

File: api/envoy/api/v2/rds.proto [hunk 1]
```diff
@@ -54,7 +54,7 @@ service RouteDiscoveryService {
 // will match an alias to a virtual host based on the content of :ref:`domains'
 // <envoy_api_msg_route.VirtualHost.domains>` field. The *resource_names_unsubscribe* field contains
 // a list of virtual host names that have been `unsubscribed
-// <https://github.com/envoyproxy/envoy/blob/master/api/XDS_PROTOCOL.md#unsubscribing-from-resources>`_
+// <https://github.com/envoyproxy/envoy/blob/master/api/xds_protocol.rst#unsubscribing-from-resources>`_
```
Comment (mattklein123) [inline api/envoy/api/v2/rds.proto]: This can be a ref link

Comment (silentdai) [reply api/envoy/api/v2/rds.proto]: Too excited about the 0 warning and didn't realize this one does't give warning. Fixed now

**Comment by mattklein123** [comment]
Oops sorry looks like this needs a format fix.

/wait

**Comment by silentdai** [comment]
This is not good. `ref` is more than spelling error. It doesn't give warning on the missing reference. Will fix 
 this one and root cause later.

**Comment by silentdai** [comment]
I realize that this piece of comment is associated with `service` instead of `message`

`service` doc is not generated, let alone validation.

**Comment by silentdai** [comment]
created https://github.com/envoyproxy/envoy/issues/7030

**Pull request merged**